modules/nrf_wifi/shim.c

```diff
@@ -27,21 +28,79 @@
 #include "qspi_if.h"
 
 LOG_MODULE_REGISTER(wifi_nrf, CONFIG_WIFI_NRF70_LOG_LEVEL);
+#if defined(CONFIG_NOCACHE_MEMORY)
+K_HEAP_DEFINE_NOCACHE(wifi_drv_ctrl_mem_pool, CONFIG_NRF_WIFI_CTRL_HEAP_SIZE);
+K_HEAP_DEFINE_NOCACHE(wifi_drv_data_mem_pool, CONFIG_NRF_WIFI_DATA_HEAP_SIZE);
+#else
+K_HEAP_DEFINE(wifi_drv_ctrl_mem_pool, CONFIG_NRF_WIFI_CTRL_HEAP_SIZE);
+K_HEAP_DEFINE(wifi_drv_data_mem_pool, CONFIG_NRF_WIFI_DATA_HEAP_SIZE);
+#endif /* CONFIG_NOCACHE_MEMORY */
```

Comment:
System heap uses cached heap `K_HEAP_DEFINE`, any reason to use uncached? In some platforms it can impact performance, like 54H.

Reply:
Since this is protected with `CONFIG_NOCACHE_MEMORY`, this should still be fine until application decides to turn on the `NOCACHE` option. Right?

Reply:
Sure, just want to make sure there are no surprises to the current behavior.